libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/ReplayLocationEngine.kt

```diff
@@ -22,7 +23,7 @@ class ReplayLocationEngine(
     mapboxReplayer: MapboxReplayer
 ) : LocationEngine, ReplayEventsObserver {
 
-    private val registeredCallbacks: MutableList<EngineCallback> = mutableListOf()
+    private val registeredCallbacks: MutableList<EngineCallback> = CopyOnWriteArrayList()
     private val lastLocationCallbacks: MutableList<EngineCallback> = mutableListOf()
```

Comment:
I think we should do the same for `lastLocationCallbacks`.

Reply:
no need as there is no API to remove an item from `lastLocationCallbacks` from the outside